### ISSUE TITLE
Pre-training stage is used to initialize backbone neural networks. It may be necessary to achieve the reported performance. Can you provide implementation details of the pre-training stage or the source code. Thanks!

### ISSUE THREAD

**Issue by haofusheng** [Sha-Lab/FEAT#1]
Pre-training stage is used to initialize backbone neural networks. It may be necessary to achieve the reported performance. Can you provide implementation details of the pre-training stage or the source code. Thanks!

**Comment by hexiang-hu** [comment]
Hi Fusheng,

We have released the pre-trained model's weight [here](https://drive.google.com/open?id=14Jn1t9JxH-CxjfWy4JmVpCxkC9cDqqfE). Our pretraining procedure is basically the same as [this](https://arxiv.org/abs/1807.05960), trained on 64 training classes and validated on 16 validation classes. Model selection is based on nearest neighbor classifier's result (accuracy) on 16 validation classes. Please refer to [this](https://github.com/pytorch/examples/tree/master/imagenet) repo for your reference of implementation details.

**Comment by ChengJiacheng** [comment]
> Hi Fusheng,
> 
> We have released the pre-trained model's weight [here](https://drive.google.com/open?id=14Jn1t9JxH-CxjfWy4JmVpCxkC9cDqqfE). Our pretraining procedure is basically the same as [this](https://arxiv.org/abs/1807.05960), trained on 64 training classes and validated on 16 validation classes. Model selection is based on nearest neighbor classifier's result (accuracy) on 16 validation classes. Please refer to [this](https://github.com/pytorch/examples/tree/master/imagenet) repo for your reference of implementation details.

Hi, hexiang

I am also planning to implement the pretraining by myself. 
The repo you list seems not applicable for training a wide residual net used in the paper
Hence, I am not sure if I understand the pretraining procedure correctly. 
Is it true that I need to add a 640*64 linear layer after the embedding (model.endoder) and then train the 64-way classifier under the cross-entropy loss?

Any information will be greatly appreciated.
Thanks!

**Comment by icoz69** [comment]
> > Hi Fusheng,
> > We have released the pre-trained model's weight [here](https://drive.google.com/open?id=14Jn1t9JxH-CxjfWy4JmVpCxkC9cDqqfE). Our pretraining procedure is basically the same as [this](https://arxiv.org/abs/1807.05960), trained on 64 training classes and validated on 16 validation classes. Model selection is based on nearest neighbor classifier's result (accuracy) on 16 validation classes. Please refer to [this](https://github.com/pytorch/examples/tree/master/imagenet) repo for your reference of implementation details.
> 
> Hi, hexiang
> 
> I am also planning to implement the pretraining by myself.
> The repo you list seems not applicable for training a wide residual net used in the paper
> Hence, I am not sure if I understand the pretraining procedure correctly.
> Is it true that I need to add a 640*64 linear layer after the embedding (model.endoder) and then train the 64-way classifier under the cross-entropy loss?
> 
> Any information will be greatly appreciated.
> Thanks!

Hi have you solved this problem? @ChengJiacheng
